src/Common/src/Interop/Windows/winhttp/Interop.SafeInternetHandle.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class WinHttp
+    {
+        // TODO: Move SafeHandleZeroOrMinusOneIsInvalid class to Common
+        internal abstract class SafeHandleZeroOrMinusOneIsInvalid : SafeHandle
+        {
+            protected SafeHandleZeroOrMinusOneIsInvalid(bool ownsHandle)
+                : base(IntPtr.Zero, ownsHandle)
+            {
+            }
+
+            public override bool IsInvalid
+            {
+                get { return this.handle == IntPtr.Zero || this.handle == (IntPtr)(-1); }
+            }
+        }
+
+        internal class SafeInternetHandle : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            public SafeInternetHandle() : base(true)
+            {
+            }
+
+            public static void DisposeAndClearHandle(ref SafeInternetHandle handle)
+            {
+                if (handle != null)
+                {
+                    handle.Dispose();
+                    handle = null;
+                }
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // TODO: Add logging so we know when the handle gets closed.
+                return Interop.WinHttp.WinHttpCloseHandle(this.handle);
```

Comment:
I actually misread the code while skimming it and thought this was the ReleaseHandle method for SafeHandleZeroOrMinusOneIsInvalid, which is why I commented on i being so general :smile: But I do like your SafeWinHttpHandle suggestion better than SafeInternetHandle, which is also general.